cfme/tests/networks/test_sdn_balancers.py

```diff
@@ -12,26 +14,35 @@
 ]
 
 
-def test_prov_balances_number(appliance):
+@pytest.fixture(scope='module')
+def network_prov_with_load_balancers(appliance):
+    prov_collection = appliance.collections.network_providers
+    providers = prov_collection.all()
+    available_prov = {}
+    for prov in providers:
+        try:
+            sum_all = len(prov.balancers.all())
+        except DestinationNotFound:
+            pytest.skip("No available load balancers for current provider")
+        available_prov[prov] = sum_all
+    return available_prov
+
+
+def test_prov_balances_number(network_prov_with_load_balancers):
     """
     Test number of balancers on 1 provider
     Prerequisites:
         Only one refreshed cloud provider in cfme database
     """
-    prov_collection = appliance.collections.network_providers
-    providers = prov_collection.all()
-    for prov in providers:
+    for prov, sum_all in network_prov_with_load_balancers.items():
         view = navigate_to(prov, 'Details')
         balancers_number = view.entities.relationships.get_text_of('Load Balancers')
-        sum_all = len(prov.balancers.all())
         assert int(balancers_number) == sum_all
 
 
-def test_balances_detail(provider, appliance):
+def test_balances_detail(provider, network_prov_with_load_balancers):
     """ Test of getting attribute from balancer object """
-    collection = appliance.collections.network_providers
-    providers = collection.all()
-    for prov in providers:
+    for prov, _ in network_prov_with_load_balancers.items():
         for balancer in prov.balancers.all():
             check = balancer.health_checks
             assert check is not None
```

Comment:
**REQUIRED:** Just to ask why are we deleting the provider below here? You should also apply an else on your for loop to skip if no providers were found else the test will Falsely pass

Reply:
Not sure why this provider delete was added, but yes it is unnecessary here.
And as for else, I'm not sure I understand as we add provider before test run, and test will skip if provider doesn't have load balancers. Or did I missed something?